### PR TITLE
Refactor: Performance tune for OASIS service

### LINKED ISSUE #345: Profile OASIS service for charge station based routing - Round 1
subtask of #344

## Initial status
Test device is mac book pro mid2014, 2.5GHZ interl Core i7, 16GB 1600 MHZ DDR3, SSD

### Profile based on North America data
Performance
| From - To   |      Distance(miles)      |  Time(seconds) | Comments
|----------|:-------------:|------:|---|
| SF to San Jose |  50  | 19  |http://127.0.0.1:8090/oasis/v1/earliest/-122.425871,37.784722;-121.554942,37.001320?max_range=500000.0&curr_range=30000.0|
| San Jose to LA |    350   |   65 |http://127.0.0.1:8090/oasis/v1/earliest/-121.876472,37.321901;-118.230489,34.064539?max_range=500000.0&curr_range=30000.0|
| Detroit to Chicago | 282 |    16 | http://127.0.0.1:8090/oasis/v1/earliest/-83.050313,42.334200;-87.684499,41.862942?max_range=500000.0&curr_range=30000.0|
| Kansas to Nashville, Tennessee | 550 |    11 |http://127.0.0.1:8090/oasis/v1/earliest/-94.587561,39.083669;-86.779198,36.161288?max_range=500000.0&curr_range=30000.0|
| Manhatten, NY to washington DC | 240 |    52 | http://127.0.0.1:8090/oasis/v1/earliest/-73.995595,40.734864;-77.038949,38.902079?max_range=500000.0&curr_range=30000.0|

Data profile

```json
{
   "count":  "comment out due to sensitive internal data source",   
   "valid_count":"comment out due to sensitive internal data source",  
   "average_nearby_ids_count":5138,
   "max_nearby_ids_count":9830,
   "min_nearby_ids_count":1,
   "average_value_max_distance":806779.9993183401,
   "max_value_max_distance":821709.1034231585,
   "min_value_max_distance":205.54475672961786,
   "average_value_max_duration":36341.44141073571,
   "max_value_max_duration":37013.92357761975,
   "min_value_max_duration":9.258772825658463,
   "maxrange_set_by_preprocessing":800000
}
```
Init time is about 30 ~ 35 seconds
```
Initialize OASIS resource manager takes 31.394521 seconds.
```

### PR DESCRIPTION
**Pull request by CodeBear801**
# Issue

- Targeting issue: the issue will be CLOSED once the PR merged. If there is no issue that addresses the problem, please open a corresponding issue and link it here. Uses the [Closes keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to close them automatically.     
Closes #345, #349 

- Any other related issue? Mention them here.    

## Description    
This pull improves performance for OASIS service for about 10 times.  For more details please go to #345 and #349.
During performance tune and refactoring, I noticed there are some further work to be done:
- Refactor structure of OASIS service https://github.com/Telenav/osrm-backend/issues/349
- Remove the name of `common`, which is not following Telenav's behavior :)     

## Tasklist

 - [x] CHANGELOG-FORK.md entry ([CHANGELOG](https://github.com/Telenav/osrm-backend/wiki/CHANGELOG))
 - [ ] [profiles/CHANGELOG.md](https://github.com/Telenav/osrm-backend/blob/master/profiles/CHANGELOG.md) entry if any `Lua` changes   
 - [ ] update relevant [Wiki pages](https://github.com/Project-OSRM/osrm-backend/wiki)
 - [ ] add tests (see [testing documentation](https://github.com/Project-OSRM/osrm-backend/blob/master/docs/testing.md)

## Pre-requirements
- Want to contribute? Great! First, please read this page [Contribution Guidelines](https://github.com/Telenav/osrm-backend/wiki/Contribution-Guidelines).    
- If your PR is still work in progress please attach the relevant label.

### CODE REVIEW

File: integration/service/oasis/connectivitymap/builder.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package connectivitymap
 import (
 	"sync"
 
+	"github.com/Telenav/osrm-backend/integration/service/oasis/internal/common"
```
Comment (wangyoucao577) [inline integration/service/oasis/connectivitymap/builder.go:6]: Still common? Aha

File: integration/service/oasis/stationfinder/stationfinderalg/querier_impl_by_stationfinder.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+	return querier
+}
+
+// NearByStationQuery finds near by stations by given placeID and return them in recorded sequence
+// Returns nil if given placeID is not found or no connectivity
+func (q *querier) NearByStationQuery(placeID common.PlaceID) []*common.RankedPlaceInfo {
+	if results, ok := q.id2QueryResults[placeID]; ok {
+		return results
+	} else {
```
Comment (wangyoucao577) [inline integration/service/oasis/stationfinder/stationfinderalg/querier_impl_by_stationfinder.go:77]: Better to remove `else` block.

File: integration/service/oasis/stationgraph/station_graph.go [hunk 3]
```diff
@@ -31,36 +31,33 @@ func NewStationGraph(currEnergyLevel, maxEnergyLevel float64, strategy chargings
 func (sg *stationGraph) setStartAndEndForGraph(currEnergyLevel, maxEnergyLevel float64) bool {
 	startLocation := sg.querier.GetLocation(stationfindertype.OrigLocationID)
 	if startLocation == nil {
-		glog.Errorf("Failed to find %#v from Querier GetLocation()\n", stationfindertype.OrigLocationID)
+		glog.Errorf("Failed to find %#v from Querier GetLocation()\n", stationfindertype.OrigLocationID.String())
```
Comment (wangyoucao577) [inline integration/service/oasis/stationgraph/station_graph.go:34]: Use `%s` instead, then you don't need to explicitly call `String()`.

Comment (CodeBear801) [reply integration/service/oasis/stationgraph/station_graph.go:34]: Agree, remove .String()

File: integration/service/oasis/connectivitymap/builder.go (hunk 1 above)
Comment (CodeBear801) [reply integration/service/oasis/connectivitymap/builder.go:6]: Will change it to `domain/entity` :)

File: integration/service/oasis/stationfinder/stationfinderalg/querier_impl_by_stationfinder.go (hunk 2 above)
Comment (CodeBear801) [reply integration/service/oasis/stationfinder/stationfinderalg/querier_impl_by_stationfinder.go:77]: Yes.  Remove not necessary `else`.

**Pull request merged**